dashboard/app/models/levels/level.rb

```diff
@@ -844,6 +844,7 @@ def summarize_for_lab2_properties(script, script_level = nil, current_user = nil
     properties_camelized[:appName] = game&.app
     properties_camelized[:useRestrictedSongs] = game.use_restricted_songs?
     properties_camelized[:usesProjects] = try(:is_project_level) || channel_backed?
+    properties_camelized[:hocFinishUrl] = script.hoc_finish_url if script.hoc?
```

Comment:
Should we use `script.finish_url` or even `ScriptLevelsHelper#script_completion_redirect` so this works for non HOC progressions too? Looks like the helper method automatically checks if the congrats page can be shown or not, so that could be handy.

There's also a bigger question here of if we should be using the `redirect` field that's returned as part of the milestone response whenever progress is reported to the server (I believe that's what legacy labs use to determine which level to go to next). However, since we don't reload the page between levels we probably don't need the `redirect` URL on intermediate levels, and would only need it for the final level in which case I think `finish_url` or `script_completion_redirect` should be sufficient

Reply:
Nice sleuthing!  `ScriptLevelsHelper#script_completion_redirect` looks good.